### PR TITLE
Promote OpenHarmony targets to tier 2

### PR DESCRIPTION
**Pull request by Amanieu**
MCP: rust-lang/compiler-team#719

### CODE REVIEW

**Review by davidtwco — APPROVED** [review]
Policy compliance and rustc book changes look good to me, needs an infra sign-off.

**Comment by davidtwco** [comment]
r? infra

**Comment by matlu** [comment]
> r? infra

@davidtwco @shepmaster Do you know if anyone from infra is looking at this?

**Comment by Amanieu** [comment]
r? infra

**Comment by Amanieu** [comment]
r? Kobzol

File: src/ci/docker/host-x86_64/dist-ohos/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    cmake \
+    sudo \
+    gdb \
+    libssl-dev \
+    pkg-config \
+    xz-utils \
+    unzip \
+    && rm -rf /var/lib/apt/lists/*
+
+COPY ohos-sdk-windows_linux-public.tar.gz /tmp
```
Comment (Kobzol) [inline src/ci/docker/host-x86_64/dist-ohos/Dockerfile:22]: This line should probably be removed, as the file is downloaded on line 24? The file is not in the repo, so the `COPY` command will most likely fail.

**Review by Kobzol — COMMENTED** [review]
CI looks good to me (left one comment regarding the Dockerfile). Could you please push a temporary commit that configures this CI workflow for `try` and run a try build? So that we can see how long it takes and if the workflow works.

File: src/ci/docker/host-x86_64/dist-ohos/Dockerfile (hunk 1 above)
Comment (Amanieu) [reply src/ci/docker/host-x86_64/dist-ohos/Dockerfile:22]: Oops, that was a leftover from testing. I'll set up a CI job to test this.

**Comment by Kobzol** [comment]
~30 minutes, that's fine. Now let's try with cache.

@bors try

**Comment by Kobzol** [comment]
24 minutes with Docker cached, nice. 

LGTM for me, I'll ask in the infra team Zulip if there are any concerns, but I think it shoud be good to go.

**Comment by Kobzol** [comment]
@Amanieu You can r=me once you remove the temporary commit.

**Pull request merged**